src/output-lua.c

```diff
@@ -622,11 +652,12 @@ static OutputInitResult OutputLuaLogInitSub(ConfNode *conf, OutputCtx *parent_ct
 
     SCMutexInit(&lua_ctx->m, NULL);
 
+    BUG_ON(parent_ctx == NULL);
+    LogLuaMasterCtx *mc = parent_ctx->data;
+    BUG_ON(mc == NULL);
+
     const char *dir = "";
-    if (parent_ctx && parent_ctx->data) {
-        LogLuaMasterCtx *mc = parent_ctx->data;
-        dir = mc->path;
-    }
+    dir = mc->script_dir;
```

Comment:
This is all to show that I don't think parent_ctx and parent_ctx->data can be NULL anymore, or its not clear how we could here with them being NULL.